ddtrace/utils/install.py

```diff
@@ -0,0 +1,24 @@
+from wrapt.importer import register_post_import_hook
+
+
+def install_module_import_hook(modulename, modulehook):
+    """Installs a module import hook for a given module.
+
+    A flag is also stored and checked on the module to so that the module is
+    not instrumented more than once.
+    """
+    # wrap the module hook with an idempotence check
+    def checked_hook(module):
```

Comment:
I added the idempotence check here because it would reduce a ton of duplication and some inconsistency in our integrations. I'm not sure if there would be any edge cases to this.

I also wonder if we could use this wrapper (or any wrapper here) to check if the integration is _enabled_ or not.

Reply:
there is also the idea that we can always patch, even if tracing is disabled, it just means we don't produce any traces/spans.

but seems likely that we can check for enabled/disabled here.